unix/icmpv4_socket.ml

```diff
@@ -0,0 +1,69 @@
+open Lwt.Infix
```

Comment:
Indeed, and including it reveals a couple of needed fixes; thanks for pointing this out.